_posts/2021-09-17-the-joy-of-slow-programming.md

```diff
@@ -0,0 +1,157 @@
+
+After a quick research I picked [Scrapy](https://scrapy.org/) - a scraping
+framework in Python, supported by a commercial platform where the code can
+run. I started putting up some code together as fast as possible. Soon I
+realised that the persistence is an important topic, because we want to
+somehow know which apartment offers are new and which are not, in order to avoid
+sending the same notifications over and over again. I tried to get the code
+running on the provided infrastructure, with each passing minute getting more
+frustrated that it doesn't work yet, and more disconnected from the reality.
+
+While I was getting deeper and deeper into the programming zone, my friend was
+sitting next to me, getting more and more bored, because at some point I
+stopped explaining what I'm currently doing. Not exactly the "hacking evening"
+anyone would wish for. After my frustration reached a certain level and my energy
+dropped to zero I did `rm -rf the-project` and told my friend: "Well, I guess
+that I need to think about it more in the next days".
+
+### New approach - take it slow
+
+The result of this thinking were a few guiding ideas that I decided on:
+1. use ["boring" technologies](/blog/2018/11/22/boring-ruby-code/) that I already
+know - Ruby, Nokogiri, Rails, and Heroku.
+2. no more than one hour of work on the project per day
+3. focus on delivering the smallest possible thing that actually provides some
+value to my friend
+4. strive for simplicity from the technical perspective and not perfection
+
+I thought about them as defining my new mindset of "slow programming". And this
+is how [Neue Wohnung](https://github.com/adamniedzielski/neue_wohnung) was born.
+If you look at the commits in the reverse order you can see that I started with
+the implementation, added tests soon, but not immediately, and configured CI
+after I had a running version in production.
+
+I was working at a slow, steady pace, trying to enjoy the every moment of it.
+As I removed the pressure of time from myself, I could actually enjoy the
+little details like a [new Rubocop rule that I didn't know before](https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Lint/ConstantDefinitionInBlock)
+or read up on [libraries that I was considering to use](https://github.com/bblimke/webmock).
+
+After a few days of working I had the first version and could actually start
+sending notifications to my friend. This meant that I was receiving feedback
+early and could adjust the application based on it, without making too many
+false assumptions.
+
+Was the fast "hacking evening" with Python a wasted time? No, not really.
+Thanks to this session I learned that Telegram is a messaging app where it's
+extremely easy to send messages programmatically and how to create my own
+Telegram bot. From the time perspective I think about it as prototyping session,
+although during the actual session I didn't have this prototyping mindset.
+
+### Next boring steps
```

Comment:
Amazing headline (in this world of click-bait and attention-grabbing everything, I absolutely mean it) :sparkles: :)

Reply:
Haha, boring can be good, like boring Ruby code 😉